### PR TITLE
[pkp-lib] #2952 unbind area from navigation menu

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: classes/navigationMenu/NavigationMenuDAO.inc.php [hunk 1]
```diff
@@ -87,6 +87,23 @@ function getByArea($contextId, $areaName) {
...
+		$params[] = (int) $contextId;
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE area_name = ? and context_id = ?',
+			$params
+		);
+
+		return new DAOResultFactory($result, $this, '_fromRow');
+	}
+
+	/**
```
Comment (NateWr) [inline classes/navigationMenu/NavigationMenuDAO.inc.php]: I don't think you need a new method in the DAO. You should be able to use `getByArea()` for the validation.

File: controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php [hunk 2]
```diff
@@ -207,6 +206,13 @@ function validate() {
 			$this->addError('path', __('manager.navigationMenus.form.duplicateTitle'));
 		}
 
+		$navigationMenu = $navigationMenuDao->getById($this->_navigationMenuId);
+
+		$navigationMenusWithArea = $navigationMenuDao->getAllByArea($this->_contextId, $this->getData('areaName'))->toArray();
+		if (count($navigationMenusWithArea) == 1 && $this->getData('areaName') != '' && $navigationMenusWithArea[0]->getId() != $this->_navigationMenuId) {
+			$this->addError('areaName', __('manager.navigationMenus.form.lastOfArea'));
+		}
+
```
Comment (NateWr) [inline controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php]: Check `$this->getData('areaName')` before making the DAO call so we can save a DB request if possible:

```
if ($this->getData('areaName') !== '') {
	$assignedMenu = $navigationMenuDao->getByArea($this->_contextId, $this->getData('areaName'));
	if ($assignedMenu && $assignedMenu->getId() !== $navigationMenu->getId()) {
		$this->addError('areaName', __('manager.navigationMenus.form.assignMultiple'));
	}
}
```

File: locale/en_US/manager.xml [hunk 3]
```diff
@@ -547,6 +547,7 @@
 	<message key="manager.navigationMenus.unassignedMenuItems">Unassigned Menu Items</message>
 	<message key="manager.navigationMenus.form.navigationMenuArea">Active Theme Navigation Areas</message>
 	<message key="manager.navigationMenus.form.navigationMenuAreaMessage">Select a navigation area</message>
+	<message key="manager.navigationMenus.form.lastOfArea">You can't bind a navigation menu to an area that is bound to another navigation menu.</message>
```
Comment (NateWr) [inline locale/en_US/manager.xml]: Update the locale key to better match what the error is, and change the text to: "A navigation menu is already assigned to this area."

File: classes/navigationMenu/NavigationMenuDAO.inc.php (hunk 1 above)
Comment (defstat) [reply classes/navigationMenu/NavigationMenuDAO.inc.php]: OK

File: controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php (hunk 2 above)
Comment (defstat) [reply controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php]: OK

File: locale/en_US/manager.xml (hunk 3 above)
Comment (defstat) [reply locale/en_US/manager.xml]: OK

File: locale/en_US/manager.xml [hunk 4]
```diff
@@ -547,6 +547,7 @@
 	<message key="manager.navigationMenus.unassignedMenuItems">Unassigned Menu Items</message>
 	<message key="manager.navigationMenus.form.navigationMenuArea">Active Theme Navigation Areas</message>
 	<message key="manager.navigationMenus.form.navigationMenuAreaMessage">Select a navigation area</message>
+	<message key="manager.navigationMenus.form.lastOfArea">A navigation menu is already assigned to this area.</message>
```
Comment (NateWr) [inline locale/en_US/manager.xml]: Let's change the `lastOfArea` key to something that better indicates the error message. Maybe `menuAssigned`?

Comment (defstat) [reply locale/en_US/manager.xml]: Changed

**Pull request merged**